tests/instruments/toptica/test_ibeamsmart.py

```diff
@@ -0,0 +1,74 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pytest
+
+from pymeasure.test import expected_protocol
+
+
+from pymeasure.instruments.toptica import IBeamSmart
+
+
+# TODO verify the talk usual communication
+init_comm = [("echo off", None), ("prom off", None), ("talk usual", ""), (None, "[OK]")]
```

Comment:
I guess this initial communication is somehow valid and likely good enough for the tests. In reality I remember that initially the device still sends back something. The exact message depends on the history of commands sent before. After a power cycle when the echo is turned on, I guess it still echos the first command and also still sends back a prompt. If this is the second time the device is opened (no power cycling in between) then the replies to these commands will be the ones also expected below (i.e. empty string + "[OK]")

If ProtocolAdapter implements the `flush_read_buffer` method I would suggest to add some replies here. Otherwise there should maybe be some extended comment here to warn not to adapt this to the real replies.

Maybe @waveman68 can provide the exact communication upon sending these commands?

Reply:
Thanks for the comment.

Yes, I expected that there is some response, which gets discarded by that flush. As the protocol does not know flush, I dropped the responses.
It's useless to define communication which is flushed anyway (at least for testing the properties).

Good idea to mention it.

Reply:
I think we already reached a conclusion here. thanks for new comment. But I can now confirm (after testing with  the hardware) that after a power cycle the first command ("echo off") still is recorded with an echo.